plugin/ale.vim

```diff
@@ -23,25 +23,21 @@ endif
 
 if !s:has_features
     " Only output a warning if editing some special files.
-    if index(['', 'gitcommit'], &filetype) == -1
-        execute 'echoerr ''ALE requires NeoVim >= 0.1.5 or Vim 8 with +timers +job +channel'''
+    if index(['', 'gitcommit'], &filetype) != -1
+        execute 'echoerr ''ALE requires NeoVim >= 0.2.5 or Vim 8 with +timers +job +channel'''
         execute 'echoerr ''Please update your editor appropriately.'''
     endif
 
     " Stop here, as it won't work.
     finish
 endif
 
-if has('nvim') && !has('nvim-0.2.0') && !get(g:, 'ale_use_deprecated_neovim')
-    execute 'echom ''ALE support for NeoVim versions below 0.2.0 is deprecated.'''
-    execute 'echom ''Use `let g:ale_use_deprecated_neovim = 1` to silence this warning for now.'''
-endif
-
 " This flag can be set to 0 to disable emitting conflict warnings.
 let g:ale_emit_conflict_warnings = get(g:, 'ale_emit_conflict_warnings', 1)
 
-if g:ale_emit_conflict_warnings
-\&& match(&runtimepath, '[/\\]ale[/\\]after') < 0
+if g:ale_emit_conflict_warnings &&
+\ matchstr(&runtimepath, '/ale/after') is# ''
+\ && matchstr(&runtimepath, '\\ale\\after') is# ''
     " Add the after directory to the runtimepath
     " This is only done if the after directory isn't already in runtimepath
     let &runtimepath .= ',' . expand('<sfile>:p:h:h') . '/after'
```

Comment:
I think we can just remove the conflict warnings stuff now. I don't think we need it any more. If we get issues about conflicts, we can just document them and leave some comments.